### PR TITLE
parallel-full: fix missing files

### PR DESCRIPTION
**Pull request by amarshall**
Previously, this was not usable as a drop-in replacement for `parallel` since it only included the one exe. This changes to additionally symlink all files from $out as well as symlink all other outputs. So no files are missing compared to `parallel`, while being efficiently symlinked.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 304018` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>parallel-full</li>
    <li>parallel-full.doc</li>
    <li>parallel-full.man</li>
  </ul>
</details>

Looking very good, properly propagates `meta` and the other outputs

File: pkgs/tools/misc/parallel/wrapper.nix [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-    --set PERL5LIB "${perlPackages.makeFullPerlPath extraPerlPackages}" \
-    ${lib.optionalString willCite "--add-flags --will-cite"}
-''
+symlinkJoin {
+  name = "parallel-full-${parallel.version}";
+  inherit (parallel) pname version meta outputs;
+  nativeBuildInputs = [ makeWrapper ];
+  paths = [ parallel ];
+  postBuild = ''
+    ${lib.concatMapStringsSep "\n" (output: "ln -s ${parallel.${output}} \$${output}") parallel.outputs}
```
Comment (pbsds) [inline pkgs/tools/misc/parallel/wrapper.nix:11]: my only snag is that `parallel.outputs` includes `out`, which i'd like to be filtered

**Comment by tomberek** [comment]
Something doesn't look right:
```
[nix-shell:~/.cache/nixpkgs-review/pr-304018]$ ls -alh ./results/parallel-full/
total 387M
dr-xr-xr-x     4 root root      5 Dec 31  1969 .
drwxrwxr-t 52161 root nixbld 662K May  4 02:43 ..
dr-xr-xr-x     2 root root     22 Dec 31  1969 bin
lrwxrwxrwx     1 root root     61 Dec 31  1969 hsx8w57vggjjvhfmhyr3aslz3j6vvlp0-parallel-20240322 -> /nix/store/hsx8w57vggjjvhfmhyr3aslz3j6vvlp0-parallel-20240322
dr-xr-xr-x     4 root root      4 Dec 31  1969 share
```
As if something was symlinked with the store path hash.

**Comment by pbsds** [comment]
One week of radio silence, I'll assume review. I both manually inspected and ran a `diff -r` between the wrapped and unwrapped derivation, only the wrapper and the `sym` symlink are different. LGTM, thanks! 

Result of `nixpkgs-review pr 304018` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>parallel-full</li>
    <li>parallel-full.doc</li>
    <li>parallel-full.man</li>
  </ul>
</details>

**Pull request merged**